jade-tui

```diff
@@ -0,0 +1,212 @@
+disk_dev=$(lsblk -pdo name | grep -v zram | grep -v NAME | grep -v loop | grep -v sr | gum choose --limit 1)
+disk=$(echo ${disk_dev} | awk '{ print substr ($0, 6 ) }')
+
+# Select desktop
+clear
+gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Select a desktop to use"
+desktop=$(gum choose --limit 1 gnome kde budgie mate cinnamon lxqt sway i3gaps herbstluftwm awesome bspwm)
+
+# misc settings
+clear
+gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Some miscellaneous settings" "Use space to enable/disable"
+misc_settings=$(gum choose --limit 4 "Enable ipv6" "Enable timeshift" "Enable zramd" "Enable flatpak")
+enable_ipv6="false"
+enable_timeshift="false"
+enable_zramd="false"
+enable_flatpak="false"
+if [[ $misc_settings == *"ipv6"* ]]; then
+	enable_ipv6="true"
+fi
+if [[ $misc_settings == *"timeshift"* ]]; then
+	enable_timeshift="true"
+fi
+if [[ $misc_settings == *"zramd"* ]]; then
+	enable_zramd="true"
+fi
+if [[ $misc_settings == *"flatpak"* ]]; then
+	enable_flatpak="true"
+fi
+
+# Check if system is uefi
+is_uefi=$([ -d /sys/firmware/efi ] && echo true || echo false)
+if [[ $is_uefi == "true" ]]; then
+ 	grub_type="grub-efi"
+	grub_location="/boot/efi"
+else
+	grub_type="grub-legacy"
+	grub_location="${disk_dev}"
+fi
+
+# Summary
+clear
+CONTINUE=false
+gum confirm "$(gum style --border normal --margin '1' --padding '1 2' --border-foreground 212 "Summary, is this correct?" "" "keymap: ${keymap}" "timezone: ${timezone}" "locale: ${locale}" "username: ${username}" "password: ${password}" "Default shell: ${shell}" "root-password: ${root_password}" "hostname: ${hostname}" "disk: ${disk}" "desktop: ${desktop}" "ipv6: ${enable_ipv6}" "timeshift: ${enable_timeshift}" "enable zramd: ${enable_zramd}" "enable flatpak: ${enable_flatpak}" "efi: ${is_uefi}")" && CONTINUE=true
+if [[ $CONTINUE != "true" ]]; then
+	echo "Exiting. Have a good day!"
+	exit
+else
+	# Remove config.json if it exists
+    if [[ $(ls | grep "/tmp/config.json") ]]; then
+        rm /tmp/config.json
+    fi
+	# Make config.json
+	echo "{
+		\"partition\": {
+			\"device\": \"${disk}\",
+			\"mode\": \"Auto\",
+			\"efi\": ${is_uefi},
+			\"partitions\": []
+		},
+		\"bootloader\": {
+			\"type\": \"${grub_type}\",
+			\"location\": \"${grub_location}\"
+		},
+		\"locale\": {
+			\"locale\": [
+				\"${locale}\"
+			],
+			\"keymap\": \"${keymap}\",
+			\"timezone\": \"${timezone}\"
+		},
+		\"networking\": {
+			\"hostname\": \"${hostname}\",
+			\"ipv6\": ${enable_ipv6}
+		},
+		\"users\": [
+			{
+				\"name\": \"${username}\",
+				\"password\": \"${crypt_password}\",
+				\"hasroot\": true,
+				\"shell\": \"${shell}\"
+			}
+		],
+		\"rootpass\": \"${crypt_root_password}\",
+		\"desktop\": \"${desktop}\",
+		\"timeshift\": ${enable_timeshift},
+		\"extra_packages\": [
+            \"fish\"
```

Comment:
Please remove this extra package, as not every user wants/needs fish on their system